python-avd/pyavd/schema/metaschema/meta_schema_model.py

```diff
@@ -0,0 +1,559 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from abc import ABC
+from enum import Enum
+from functools import cached_property
+from typing import Annotated, Any, ClassVar, Generator, List, Literal
+
+from pydantic import BaseModel, ConfigDict, Field, constr
+
+from ..generate_docs.tablerowgen import TableRow, TableRowGenBase, TableRowGenBool, TableRowGenDict, TableRowGenInt, TableRowGenList, TableRowGenStr
+from ..generate_docs.yamllinegen import YamlLine, YamlLineGenBase, YamlLineGenBool, YamlLineGenDict, YamlLineGenInt, YamlLineGenList, YamlLineGenStr
+from .resolvemodel import merge_schema_from_ref
+
+"""
+This module provides Pydantic models (classes) representing the meta-schema of the AVD Schema.
+
+Each variable in the schema is called a field, and for each type of field we have a corrosponding Pydantic model:
+- AvdSchemaInt
+- AvdSchemaBool
+- AvdSchemaStr
+- AvdSchemaList
+- AvdSchemaDict
+
+The alias "AvdSchemaField" is a union of of all the models above, and can be used as easy type hint for any field type.
```

Comment:
Do you need this or could you simply use `AvdSchemaBaseModel` instead (though I agree the name may be better with Field)

Reply:
I prefer to keep the alias so we get type checking of that too.